rust_src/src/list.rs

```diff
@@ -0,0 +1,37 @@
+use std::os::raw::c_char;
+use std::ptr;
+
+extern crate libc;
+
+use lisp::{LispObject, LispSubr, Qnil, Qt, VectorLikeHeader, PvecType, PSEUDOVECTOR_AREA_BITS};
+use cons::{CONSP, NILP};
+
+fn Flistp(object: LispObject) -> LispObject {
+    if CONSP(object) || NILP(object) {
+        unsafe { Qt }
+    } else {
+        Qnil
+    }
+}
+
+defun!("listp",
+       Flistp,
+       Slistp,
+       1, 1,
+       ptr::null(),
+       "Return t if OBJECT is a list, that is, a cons cell or nil. Otherwise, return nil.");
```

Comment:
It's probably worth adding `(fn OBJECT)` at the end of the docstring, or `*Help*` will just show `(listp ARG)`.